src/main/java/com/configcat/ConfigCatClient.java

```diff
@@ -531,34 +531,34 @@ private String getVariationIdFromSettingsMap(SettingResult settingResult, String
         try {
             Map<String, Setting> settings = settingResult.settings();
             if (settings.isEmpty()) {
-                String error = "Config JSON is not present. Returning defaultVariationId: [" + defaultVariationId + "].";
+                String error = "Config JSON is not present. Returning the `defaultVariationId` parameter that you specified in your application: '" + defaultVariationId + "'.";
                 this.hooks.invokeOnFlagEvaluated(EvaluationDetails.fromError(key, null, error, userObject));
-                this.logger.error(error);
+                this.logger.error(1000, errorMessage);
                 return defaultVariationId;
             }
 
             Setting setting = settings.get(key);
             if (setting == null) {
-                String error = "Variation ID not found for key " + key + ". Here are the available keys: " + String.join(", ", settings.keySet());
+                String error = "Failed to evaluate setting '" + key + "' (the key was not found in config JSON). Returning the `defaultVariationId` parameter that you specified in your application: '" + defaultVariationId + "'. Available keys: " + String.join(", ", settings.keySet()) + ".";
```

Comment:
Available keys should be surrounded with single quotes like `'Key1', 'Key2', 'Key3'`